frontend/src/index.tsx

```diff
@@ -1,10 +1,20 @@
 import * as React from "react";
-import {render} from "react-dom";
 import App from "./components/App";
+import { CLIENT_ID } from "./services/authentication_handler_service/authentication_constants";
+import { render } from "react-dom";
+import { Unloaded } from "./components/pages/unloaded/Unloaded";
 
 const rootEl = document.getElementById("root");
 
-render(
-    <App/>,
-    rootEl,
-);
+// Perform initialization steps with gapi.
+window.gapi.load('auth2', () => {
+  window.gapi.auth2.init({
+    client_id: CLIENT_ID,
+  }).then(() => {
+    // Upon the initialization of Auth2, render the page.
+    render(<App/>, rootEl);
+  });
+});
```

Comment:
while in an ideal world we shouldn't modify index.tsx, I agree that this was the right call to perform the initialization before App is rendered.

Reply:
Yeah I went back and forth on this, only touching app.tsx would have been nice but the gapi configuration code was out of place and even in a useEffect with empty dependencies would run at least twice.

Reply:
Yeah definitely best to avoid in general but seems reasonable in this case.